### PR TITLE
Update Forseti inventory storage to store the temporary CAI asset data.

### PR DESCRIPTION
**Pull request by ahoying**
* Add CAI support to server configuration.

Thanks for opening a Pull Request!

Here's a handy checklist to ensure your PR goes smoothly.

- [x] I signed Google's [Contributor License Agreement](https://opensource.google.com/docs/cla/)
- [x] My code conforms to Google's [python style](https://google.github.io/styleguide/pyguide.html).
- [x] My PR at a minimum doesn't decrease unit-test coverage (if applicable).
- [x] My PR has been functionally tested.
- [x] All of the [unit-tests](https://forsetisecurity.org/docs/latest/develop/dev/testing.html) still pass.
- [x] Running `pylint --rcfile=pylintrc` passes.

These guidelines and more can be found in our [contributing guidelines](https://github.com/GoogleCloudPlatform/forseti-security/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ahoying** [comment]
Sorry for the few unrelated test changes, debugging the .egg issue I was having with the google-cloud-cloudasset package lead to discovering some other broken things in the code base. If you'd like me to move those to a different PR, let me know.

**Comment by ahoying** [comment]
Also the decrease in coverage is from fixing tests that were broken, so they are no longer 'testing' some error paths in the code accidentally. Those error paths are now untested, but should have specific tests created instead of accidentally getting run.

File: google/cloud/forseti/services/inventory/storage.py [hunk 1]
```diff
@@ -636,6 +761,91 @@ def _get_resource_id(self, resource):
...
+            asset_type (str): The asset type to return
+            name (str): The resource to return.
+
+        Returns:
+            dict: The content data for the specified resource.
+        """
+        if not self.has_cai_data:
+            return {}
+
+        row = self.session.query(CaiTemporaryStore).filter(
```
Comment (joecheuk) [inline google/cloud/forseti/services/inventory/storage.py]: qq: Is there any reason why we are using 2 diff styles to do filtering (compare with above)?

> 

        filters = [
            CaiTemporaryStore.content_type == content_type,
            CaiTemporaryStore.asset_type == asset_type,
            CaiTemporaryStore.parent_name == parent_name,
        ]
        base_query = self.session.query(CaiTemporaryStore)
         for qry_filter in filters:
            base_query = base_query.filter(qry_filter)

File: google/cloud/forseti/services/inventory/storage.py [hunk 2]
```diff
@@ -636,6 +761,91 @@ def _get_resource_id(self, resource):
...
         return 0
 
+    def clear_cai_data(self):
+        """Deletes all temporary CAI data from the cai temporary table.
+
+        Returns:
+            int: The number of rows deleted.
+        """
+        num_rows = CaiTemporaryStore.delete_all(self.session)
+        self.has_cai_data = False
```
[the remark this answers is not in the document]
Comment (ahoying) [reply google/cloud/forseti/services/inventory/storage.py]: Good point, thanks.

File: google/cloud/forseti/services/inventory/storage.py [hunk 3]
```diff
@@ -636,6 +761,91 @@ def _get_resource_id(self, resource):
...
+        """
+        num_rows = 0
+        for line in data.split('\n'):
+            if not line:
+                continue
+            row = CaiTemporaryStore.from_json(line)
+            self.buffer.add(row)
+            num_rows += 1
+        self.buffer.flush()
+        self.session.commit()
```
[the remark this answers is not in the document]
Comment (ahoying) [reply google/cloud/forseti/services/inventory/storage.py]: Yep, good idea, fixed.

File: google/cloud/forseti/services/inventory/storage.py (hunk 1 above)
Comment (ahoying) [reply google/cloud/forseti/services/inventory/storage.py]: You're right, fixed.

File: google/cloud/forseti/services/inventory/storage.py [hunk 4]
```diff
@@ -399,6 +525,8 @@ def flush(self):
 
         self.session.add_all(self.buffer)
         self.session.flush()
+        if self.commit_on_flush:
+            self.session.commit()
```
[the remark this answers is not in the document]
Comment (ahoying) [reply google/cloud/forseti/services/inventory/storage.py]: Yep, although I kept the exemption handling outside of this class to remain consistent.

**Pull request merged**